tasks/main.yml

```diff
@@ -1,2 +1,209 @@
 ---
 
+- name: Install postfix
+  package:
+    name:
+      - postfix
+      - postfix-ldap
+      - postfix-pgsql
+    state: present
+
+- name: Setup LDAP authentication
+  block:
+    - name: Install saslauthd
+      package:
+        name:
+          - libsasl2-modules
+          - sasl2-bin
+        state: present
+
+    - name: Place LDAP connection config file
+      template:
+        src: saslauthd.conf.j2
+        dest: /etc/saslauthd.conf
+
+    - name: Place SMTP SASL config file
+      copy:
+        src: smtpd.conf
+        dest: /etc/postfix/sasl/smtpd.conf
+
+    - name: Place saslauthd-postfix config file
+      copy:
+        src: saslauthd-postfix
+        dest: /etc/default/saslauthd-postfix
+
+    - name: Create socket directory for saslauthd
+      shell: dpkg-statoverride --add root sasl 755 /var/spool/postfix/var/run/saslauthd
+      register: dpkg_sasl
+      changed_when: "'already exists' not in dpkg_sasl.stderr"
+      failed_when: "dpkg_sasl.rc != 0 and 'already exists; aborting' not in dpkg_sasl.stderr"
+
+    - name: Add group sasl to user postfix
+      user:
+        name: postfix
+        append: True
+        groups: sasl
+
+  when: postfix_sasl_enabled is defined
+  notify: Restart saslauthd service
+
+- name: Disable and stop saslauthd
+  block:
+    - name: Install python-apt
+      package:
+        name:
+          - python-apt-dev
+        state: present
+
+    - name: Gather package facts
+      package_facts:
+        manager: auto
+
+    - name: Stop and disable saslauthd service
+      service:
+        name: saslauthd
+        state: stopped
+        enabled: no
+      when: "'libsasl2-modules' in ansible_facts.packages"
+  when: postfix_sasl_enabled is not defined
+
+- name: Setup OpenDKIM
+  block:
+    - name: Install OpenDKIM
+      package:
+        name:
+          - opendkim
+          - opendkim-tools
+        state: present
+
+    - name: Create opendkim config directory
+      file:
+        state: directory
+        dest: /etc/opendkim
+
+    - name: Create opendkim key directory
+      file:
+        state: directory
+        dest: /etc/opendkim/keys
+
+    - name: Create socket directory for opendkim
+      file:
+        state: directory
+        dest: /var/spool/postfix/var/run/opendkim
+        group: opendkim
+        mode: 0775
+
+    - name: Place opendkim config
+      copy:
+        content: "{{ postfix_opendkim_config }}"
```

Comment:
Do you mean in the readme or in the template (`| mandatory`)? The latter is not necessary, because in this context ansible throws an error when an undefined variable is used in a template.

> The task includes an option with an undefined variable. The error was: 'postfix_opendkim_config' is undefined. 'postfix_opendkim_config' is undefined

Reply:
Ok. I did not know that.